shell/components/ResourceDetail/Masthead.vue

```diff
@@ -148,7 +149,7 @@ export default {
         return this.value.namespaceLocation || {
           name:   'c-cluster-product-resource-id',
           params: {
-            cluster:  this.$route.params.cluster,
+            cluster:  this.$route.params.cluster === BLANK_CLUSTER ? this.value?.localCluster?.id : this.$route.params.cluster,
```

Comment:
we should do this via the provisioning cluster models `namespaceLocation` getter

Reply:
Agree, define a `namespaceLocation` in provisioning model is better.

Reply:
this line can now be removed (given we should have `value.namespaceLocation`)